src/Access/Authentication.cpp

```diff
@@ -37,14 +39,17 @@ Authentication::Digest Authentication::getPasswordDoubleSHA1() const
         case DOUBLE_SHA1_PASSWORD:
             return password_hash;
 
+        case LDAP_SERVER:
+            throw Exception("Cannot get password double SHA1 for user with 'LDAP_SERVER' authentication.", ErrorCodes::BAD_ARGUMENTS);
+
         case MAX_TYPE:
             break;
     }
     throw Exception("getPasswordDoubleSHA1(): authentication type " + toString(type) + " not supported", ErrorCodes::NOT_IMPLEMENTED);
 }
 
 
-bool Authentication::isCorrectPassword(const String & password_) const
+bool Authentication::isCorrectPassword(const String & password_, const String & user_, const ExternalAuthenticators & external_authenticators) const
```

Comment:
It seems it's better to make `std::shared_ptr<const ExternalAuthenticators>` a part of the class `Authentication`. That way we will need to pass only one argument `password` to `Authentication::isCorrectPassword()` making the interface simplier.